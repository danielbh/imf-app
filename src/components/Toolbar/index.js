import React from 'react';
import PropTypes from 'prop-types';

import ButtonBar from '../../components/ButtonBar'
import Flexbox from '../../components/Flexbox';
import ChartTabs from '../../containers/ChartTabs';

const Toolbar = () => (
  <Flexbox justifyContent="space-around">
       <ChartTabs />
       <ButtonBar />
  </Flexbox>
);

export default Toolbar
