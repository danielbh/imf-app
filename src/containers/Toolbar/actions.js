/**
 * Created by danielhollcraft on 9/8/17.
 */

import {
  SET_WEEK_RANGE,
  SET_MONTH_RANGE,
  SET_THREE_MONTH_RANGE,
  SET_YEAR_RANGE,
  SET_ALL_RANGE
} from './constants'

export const selectWeek = () => ({
  type: SET_WEEK_RANGE
});

export const selectMonth = () => ({
  type: SET_MONTH_RANGE
});

export const selectThreeMonth = () => ({
  type: SET_THREE_MONTH_RANGE
});

export const selectYear = () => ({
  type: SET_YEAR_RANGE
});

export const selectAll = () => ({
  type: SET_ALL_RANGE
});