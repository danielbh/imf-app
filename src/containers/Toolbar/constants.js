/**
 * Created by danielhollcraft on 9/8/17.
 */
export const SET_WEEK_RANGE = 'SET_WEEK_RANGE';
export const SET_MONTH_RANGE = 'SET_MONTH_RANGE';
export const SET_THREE_MONTH_RANGE = 'SET_THREE_MONTH_RANGE';
export const SET_YEAR_RANGE = 'SET_YEAR_RANGE';
export const SET_ALL_RANGE = 'SET_ALL_RANGE';
