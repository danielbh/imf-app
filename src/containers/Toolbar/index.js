import React from 'react';
import PropTypes from 'prop-types';
import { ButtonToolbar, ButtonGroup, Button } from 'react-bootstrap';
import styled from 'styled-components';
import { connect } from 'react-redux';
import {
  setDateRange
} from './actions'

import {
  WEEK,
    MONTH,
    THREE_MONTHS,
    YEAR,
    ALL
} from "./constants";

const Wrapper = styled.div`
  .toolbar {
    display: -webkit-flex;
    display: flex;
    -webkit-flex-direction: row;
    flex-direction: row;
    -webkit-justify-content: flex-end;
    justify-content: flex-end;
  }
`;

export const Toolbar = ({
  selectWeek,
  selectMonth,
  selectThreeMonth,
  selectYear,
  selectAll
}) => (
  <Wrapper>
    <ButtonToolbar className="toolbar">
      <ButtonGroup>
        <Button onClick={selectWeek}>1W</Button>
        <Button onClick={selectMonth}>1M</Button>
        <Button onClick={selectThreeMonth}>3M</Button>
        <Button onClick={selectYear}>1Y</Button>
        <Button onClick={selectAll}>All</Button>
      </ButtonGroup>
    </ButtonToolbar>
  </Wrapper>
);

Toolbar.propTypes = {
  selectWeek: PropTypes.func,
  selectMonth: PropTypes.func,
  selectThreeMonth: PropTypes.func,
  selectYear: PropTypes.func,
  selectAll: PropTypes.func
};

export function mapDispatchToProps(dispatch) {
  return {
    selectWeek: () => dispatch(setDateRange(WEEK)),
    selectMonth: () => dispatch(setDateRange(MONTH)),
    selectThreeMonth: () => dispatch(setDateRange(THREE_MONTHS)),
    selectYear: () => dispatch(setDateRange(YEAR)),
    selectAll: () => dispatch(setDateRange(ALL))
  };
}

Toolbar.defaultProps = {};

export default connect(null, mapDispatchToProps)(Toolbar);
