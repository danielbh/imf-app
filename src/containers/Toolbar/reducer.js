/**
 * Created by danielhollcraft on 9/8/17.
 */
import moment from 'moment'

import {
  WEEK,
  MONTH,
  THREE_MONTHS,
  YEAR
} from "./constants";

export const toolbar = (state = [], action) => {

  const returnEntriesInRange =
    (units, thresh = 1) => state.filter(e => moment(moment()).diff(e.date, units) < thresh);

  switch(action) {
    case WEEK:
      return returnEntriesInRange('weeks');
    case MONTH:
      return returnEntriesInRange('months');
    case THREE_MONTHS:
      return returnEntriesInRange('months', 3);
    case YEAR:
      return returnEntriesInRange('years');
    default:
      return state;
  }
};