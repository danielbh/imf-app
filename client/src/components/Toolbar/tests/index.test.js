/**
 * Created by danielhollcraft on 9/8/17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import Toolbar from '../index'

describe('<Toolbar />', () => {
  it('matches snapshot', () => {
    expect(shallow(<Toolbar />)).toMatchSnapshot();
  });
});