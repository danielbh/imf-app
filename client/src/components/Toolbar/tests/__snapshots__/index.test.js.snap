// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> matches snapshot 1`] = `
<Flexbox
  justifyContent="space-around"
>
  <Connect(Component) />
  <RangeButtons />
</Flexbox>
`;
